pkg/autohealing/controller/controller.go

```diff
@@ -327,17 +328,30 @@ func (c *Controller) repairNodes(unhealthyNodes []healthcheck.NodeInfo) {
 				// Cordon the nodes before repair.
 				for _, node := range unhealthyNodes {
 					nodeName := node.KubeNode.Name
-					newNode := node.KubeNode.DeepCopy()
-					newNode.Spec.Unschedulable = true
 
 					// Skip cordon for master node
 					if !node.IsWorker {
 						continue
 					}
-					if _, err := c.kubeClient.CoreV1().Nodes().Update(context.TODO(), newNode, metav1.UpdateOptions{}); err != nil {
-						log.Errorf("Failed to cordon node %s, error: %v", nodeName, err)
-					} else {
-						log.Infof("Node %s is cordoned", nodeName)
+					retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+						// Retrieve the latest version of Node before attempting update
+						// RetryOnConflict uses exponential backoff to avoid exhausting the apiserver
+						newNode, err := c.kubeClient.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+						if err != nil {
+							log.Errorf("Failed to get node %s, error: %v before update", nodeName, err)
+						}
+						newNode.Spec.Unschedulable = true
+						if _, updateErr := c.kubeClient.CoreV1().Nodes().Update(context.TODO(), newNode, metav1.UpdateOptions{}); updateErr != nil {
+							log.Errorf("Failed in retry to cordon node %s, error: %v", nodeName, updateErr)
```

Comment:
Same remark about usage of warning comment if this is a retriable error.

Reply:
I don't have any context here that I can use